application/schemas/templates/components.py

```diff
@@ -28,8 +41,89 @@ class Component(BaseModel):
                     'a higher priority.',
     )
 
-    class Config:
-        extra = Extra.forbid
+
+class ComponentTypeHttpMethods(StrEnum):
+    """
+    Template HTTP component methods.
+    """
+    get = 'get'
+    post = 'post'
+    put = 'put'
+    patch = 'patch'
+    delete = 'delete'
+
+
+class HttpRequest(BaseModel):
+    """
+    Deploy schema for HTTP type components.
+    """
+    url: HttpUrl = Field(description='URL for HTTP deployment.', example='http://example.com/path/to/deploy')
+    method: ComponentTypeHttpMethods = Field(
+        default=ComponentTypeHttpMethods.get,
+        description='HTTP method for deployment',
+        example=ComponentTypeHttpMethods.post)
+    headers: dict | list[dict] | None = Field(
+        description='Headers for HTTP deployment', example={
+            'Content-Type': 'application/json'})
+    parameters: dict | list[dict] | None = Field(
+        description='Parameters for HTTP deployment', example=[{'param1': 'value1'}])
+
+
+class ComponentHttp(BaseModel):
+    """
+    Application component HTTP schema.
+    """
+    name: constr(min_length=1, strip_whitespace=True) = Field(description='Component Name', example='jenkins')
+    type: Literal[ComponentTypes.http]
+    enabled: bool | TemplateVariable | None = Field(description='Is component must be installed.', default=True)
+    create: HttpRequest = Field(description='Creation details for HTTP type components')
+    delete: HttpRequest = Field(description='Deletion details for HTTP type components')
+    health: HttpRequest | None = Field(description='Health check for HTTP type components')
+
+
+Components = Annotated[ComponentHelm |
+                       ComponentHttp,
+                       Field(discriminator='type')]
+
+
+class Component(BaseModel):
```

Comment:
I think this model is redundant. You can do just `components: list[Components]` instead of `components: list[Component]` in `TemplateSchema` and validation should work fine. But don't trust me check it.))

Also you after removing `class Component(BaseModel):` you can rename `Components` to `Component`.

You can play with model and validation in real time:

1. `make setup`
2. `cd application/`
3. `python`
4. `from schemas.templates.template import TemplateSchema`
5. `with open('./path/to/template.yaml', 'r') as file:`
6. `    template = TemplateSchema.parse_obj(yaml.safe_load(file))` - pay attention on indent.